tsdb/agent/db_test.go

```diff
@@ -161,22 +159,22 @@ func TestRollback(t *testing.T) {
 		numSeries     = 8
 	)
 
-	promAgentDir, err := ioutil.TempDir("", t.Name())
-	require.NoError(t, err)
-	t.Cleanup(func() {
-		require.NoError(t, os.RemoveAll(promAgentDir))
-	})
+	promAgentDir := t.TempDir()
 
 	lbls := labelsForTest(t.Name(), numSeries)
 	opts := DefaultOptions()
 	logger := log.NewNopLogger()
 	reg := prometheus.NewRegistry()
 	remoteStorage := remote.NewStorage(log.With(logger, "component", "remote"), reg, startTime, promAgentDir, time.Second*30, nil)
+	defer func() {
+		require.NoError(t, remoteStorage.Close())
+	}()
```

Comment:
Same comment about `remoteStorage` being overwritten, hence pass `remoteStorage` as an argument to this function.